### ISSUE TITLE
interlaced_symm_v2: high-side padding on the symmetry axis corrupts results when o_shape > c_shape

### ISSUE THREAD

**Issue by yszhang95** [yszhang95/tred#59]
## Summary

`interlaced_symm_v2` (`src/tred/convo.py`) silently returns **wrong** results when the output shape `o_shape` is larger than the convolution length `c_shape` **on the symmetry axis**. Total charge is conserved (it is a *mis-placement*, not a loss), so there is no error and no obvious symptom — the mirrored half of the symmetric reconstruction lands on the wrong pixels.

Over-padding the **time axis** or a **non-symmetry spatial axis** is harmless; only the flipped (symmetry) axis is affected.

> Note on scope: this surfaced while profiling the **truth-labeling pipeline (`graph_truth.py`), which lives on a dev branch, not on `master`**. But the defect itself is in the core convolution on `master` (`src/tred/convo.py`) and also affects `graph_effq.fullsim`: `runit` hardcodes `o_shape=(8, 8, 512*4)`, which equals `c_shape` only for the 5×5-pixel response. Any response with a smaller footprint (e.g. the 3×3 response → `c_shape` spatial = 6) is convolved incorrectly.

## Root cause

The reconstruction uses reflection symmetry on `symm_axis`:

```python
meas.real = meas.real + torch.flip(meas.imag, dims=flipdims)   # flipdims = (symm_axis+1,)
```

The laces are first padded to `o_shape` with **`zero_pad`, which appends zeros on the high side**. After the per-lace FFT, the linear-convolution result sits at indices `[0, c_shape)` of the `o_shape`-long array. The `flip` then reflects about pivot `(o_shape − 1)/2`, but the mirror data is centered at `(c_shape − 1)/2`; when `o_shape > c_shape` the mirrored lace is shifted by `o_shape − c_shape` while the forward (real) half stays put, so the two no longer align.

Because the axis is not flipped on the time / non-symmetry dimensions, high-side padding there is harmless.

The shape guard only checks `>=`, so the oversized case passes validation silently:

```python
elif any(tuple(s1 > s2 for s1, s2 in zip(c_shape, o_shape))):
    raise ValueError(...)
```

with `c_shape = dft_shape(signal//steps, response//steps) = signal_pixels + response_pixels − 1`.

The older `interlaced_symm` (v1) does **not** have this bug: it pads only to `c_shape` (via `convolve`) and flips over that natural-length output, so there is never an oversize to mis-register.

## Reproduction (self-contained, `master` library only)

```python
import torch
from tred.blocking import Block
from tred.convo import interlaced, interlaced_symm_v2, dft_shape

torch.manual_seed(0)
steps = torch.tensor([10, 10, 1]); T = 8
# 3-pixel-footprint response (30 fine), symmetric about the center of the symm axis (axis 0)
base = torch.rand(30, 30, T, dtype=torch.float64)
resp = base + base.flip(0)                      # R[k] == R[29-k]  -> valid for the symmetry trick
data = torch.zeros(1, 40, 40, T, dtype=torch.float64)
for (a, b, c) in [(15,15,2),(23,7,5),(31,34,1),(6,28,6)]:
    data[0, a, b, c] = 1.0
sig = Block(location=torch.tensor([[0, 0, 0]]), data=data)

gt = interlaced(sig, resp, steps).data[0].sum(-1)        # non-symmetric reference = ground truth
print('c_shape =', dft_shape((4,4,T), (3,3,T)))          # (6, 6, 15)
for osh in [(6,6,16), (6,6,32), (6,8,16), (8,6,16), (8,8,16)]:
    a = interlaced_symm_v2(sig, resp, steps, o_shape=osh).data[0].sum(-1)
    d = float((a[:6,:6] - gt[:6,:6]).abs().max())
    print(f'  o_shape {str(osh):12s} max|a-gt|={d:.2g}  {"OK" if d < 1e-9 else "WRONG"}')
```

Output:

```
c_shape = (6, 6, 15)
  o_shape (6, 6, 16)   max|a-gt|=7.1e-15  OK     # symm axis == c_shape
  o_shape (6, 6, 32)   max|a-gt|=3.6e-15  OK     # time over-padded  -> fine
  o_shape (6, 8, 16)   max|a-gt|=7.1e-15  OK     # non-symm spatial over-padded -> fine
  o_shape (8, 6, 16)   max|a-gt|=17       WRONG  # symm axis over-padded
  o_shape (8, 8, 16)   max|a-gt|=17       WRONG
```

i.e. the result is correct **iff** `o_shape == c_shape` on the symmetry axis. (The real nd-LAr responses behave identically: the 5×5 response has `c_shape` spatial = 8 so the hardcoded `(8,8,…)` happens to be correct; the 3×3 response has `c_shape` spatial = 6 and is computed wrong at `(8,8,…)`.)

## Impact

- Any use of `interlaced_symm_v2`/`LacedConvo` with `o_shape[symm_axis] > c_shape[symm_axis]` produces wrong currents, silently.
- With the hardcoded `o_shape=(8,8,512*4)` in `runit`, this is correct only for the 5×5 response. Runs using a smaller-footprint response (e.g. 3×3) compute physically wrong induced currents on the symmetry axis.

## Proposed fix

1. Pad the **symmetry axis with centered/symmetric padding** (the `symmetric_pad` "center"-for-response / "edge"-for-signal scheme that the v1 path effectively respects) instead of high-side `zero_pad` — or perform the reconstruction `flip` over `c_shape` rather than `o_shape`. Either keeps over-padding the time / non-symmetry axes valid (useful for choosing efficient FFT sizes).
2. Tightening the guard to require `o_shape[symm_axis] == c_shape[symm_axis]` is a stop-gap that at least turns the silent corruption into a clear error.
3. Have `runit` (`graph_effq.py`, and the dev-branch `graph_truth.py`) derive `o_shape` from the response (`= c_shape`, padded only on the time axis) rather than hardcoding `(8, 8, 512*4)`.

## Affected code

- `src/tred/convo.py`: `interlaced_symm_v2` (the reconstruction `flip` over `o_shape` + the `o_shape >= c_shape` guard); `dft_shape`; `zero_pad` (high-side append).
- `src/tred/plots/graph_effq.py` (and dev-branch `graph_truth.py`): hardcoded `LacedConvo(lacing, o_shape=(8, 8, 512*4))`.
